lib/client-api/src/story_store.ts

```diff
@@ -444,22 +451,24 @@ export default class StoryStore {
         argTypes,
         globals: this._globals,
         viewMode: this._selection?.viewMode,
-      });
+      };
+
+      return this._storyContexts[id];
     };
 
-    const unboundStoryFn: LegacyStoryFn = (context: StoryContext) => getDecorated()(context);
+    const unboundStoryFn: LegacyStoryFn = (context: StoryContext) => {
+      const result = getDecorated()(context);
+      // delete this._storyContexts[id]; // <- FIXME when to delete this?
```

Comment:
I want to delete the context after the story renders, as otherwise we'll end up with a bunch of stale story contexts in the store for previously rendered stories.

I guess it doesn't work doing it here as React may not have rendered the complete tree of JSX/elements at the point that `getDecorated()(...)` is done. Probably other frameworks may not make similar guarantees.

One idea I had is to mark the context as "used" and then to remove all used contexts at the *start* of rendering any other story. So the idea would be, if you rendered two stories at the same time, neither would used, so both contexts would be set, but the next story you rendered would clear them out. A bit convoluted but it'll probably work.

Reply:
Sounds complicated but I can't think of anything better. Is this a real problem or just a theoretical one?

Reply:
I guess theoretical. Let's remove until it isn't.

Reply:
```suggestion
```